andela/travel_tool_back
# #160157497 Setup bugsnag for error reporting

#### What does this PR do?
- It integrates _bugsnag_ with the application as a tool for error monitoring.
- It makes it mandatory to supply the application with bugsnag API key for the production, staging and development environment, and optional for the test environment.
- It makes CircleCI to run database migrations just once.
- It makes sure that the deployment scripts exit if any of the command fails.

#### Description of Task to be completed?
If there are application errors while users interact with the app, the app should be able to track these errors. Tools that can be used for this include;  Kibana, Bugsnag, Stackdriver.

##### Acceptance criteria
All automatic failures from both the backend and frontend should be sent and visible from the bugsnag dashboard.

#### How should this be manually tested?
1. Create an account at [bugsnag](https://www.bugsnag.com/).
2. Create a project, and copy it's API key.
3. Add the API key in the .env file:
`BUGSNAG_API_KEY=your-bugsnag-api-key`
4. Break something in the code, and run the application.
5. Refresh the dashboard page in bugsnag, and you shall see the error.

#### Any background context you want to provide?
If the `BUGSNAG_API_KEY` is not provided, then the application would work normally without reporting errors to bugsnag.

#### What are the relevant pivotal tracker stories?
#160157497

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-06 at 12 35 14" src="https://user-images.githubusercontent.com/18000724/45149162-97ed6980-b1d1-11e8-89c3-307bd7be7dbe.png">
